sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -2650,7 +2650,7 @@ pi_result cuda_piclProgramCreateWithSource(pi_context context, pi_uint32 count,
                                            const size_t *lengths,
                                            pi_program *program) {
   cl::sycl::detail::pi::die("cuda_piclProgramCreateWithSource not implemented");
```

Comment:
remove the die, since it makes the return unreachable

Reply:
right, will do that. should I enable printing the error msg here (like level_zero) before returning? There are some more cases in the cuda plugin where die() is called before a return stmt with an error code. might be a good idea to fix those too.

Reply:
>  should I enable printing the error msg here (like level_zero) before returning?

Why not, if they have the ways to print logs.

> There are some more cases in the cuda plugin where die() is called before a return stmt with an error code. might be a good idea to fix those too.

Let's not remove any "die"-s without making sure they (error return codes) are handled properly in SYCL RT.

Reply:
> Why not, if they have the ways to print 

there is no way right now just for printing error msg.